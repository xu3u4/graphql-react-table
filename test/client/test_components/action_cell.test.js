// import ReactShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import ActionCell from 'components/common/action_cell';

describe ('<ActionCell>', () => {
  const onAction = sinon.stub();
  const action = 'Delete';
  let actionCell;
  
  it ('render components', () => {
    actionCell = renderer.create(
      <ActionCell action={ onAction } >{ action }</ActionCell>
    );
    expect(actionCell).toMatchSnapshot();
  });

  it ('should call onAction when click', () => {
    // const renderer = new ReactShallowRenderer();
    // renderer.render(
    //   <ActionCell action={ onAction } >{ action }</ActionCell>
    // );
    // actionCell = renderer.getRenderOutput();
    // console.log(actionCell);
    actionCell = shallow(
      <ActionCell action={ onAction } >{ action }</ActionCell>
    );
    actionCell.find('button').simulate('click');
    expect(onAction.calledOnce).toBeTruthy();
  });
});