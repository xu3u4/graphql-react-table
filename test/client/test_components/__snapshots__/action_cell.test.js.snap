// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionCell> render components 1`] = `
<td>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</td>
`;
